imap_processing/cdf/imap_cdf_manager.py

```diff
@@ -12,9 +12,13 @@
 class ImapCdfAttributes(CdfAttributeManager):
     """Contains IMAP specific tools and settings for CDF management."""
 
-    def __init__(self):
+    def __init__(self, source_dir_input=None):
         """Set the path to the config directory."""
-        super().__init__(Path(__file__).parent / "config")
+        if source_dir_input is None:
+            super().__init__(Path(__file__).parent / "config")
+        else:
+            super().__init__(Path(__file__).parent / "config")
+            self.source_dir = Path(__file__).parent / source_dir_input
```

Comment:
If the `super().__init__()` method is intended to be called in both cases, this could be simplified as follows
```suggestion
        super().__init__(Path(__file__).parent / "config")
        if source_dir_input is not None:
            self.source_dir = Path(__file__).parent / source_dir_input
```
But, it isn't clear to me that the default schema files are intended to be loaded if a source_dir is specified.